Lesson20/merve2/script.js

```diff
@@ -0,0 +1,146 @@
+// 1. Create a Person class with name, age, and gender properties, and a greet() method that logs a greeting message to the console.
+// Then create a Student class that extends the Person class and adds a major property. 
+//Override the greet() method in the Student class to log a different greeting message to the console.
+// Create 2 variables, one of type Person and one of type Student and log them to console.
+
+class Person {
+ constructor (name,age,gender){
+    this._name = name;
+    this._age = age;
+    this._gender = gender;
+} 
+  greet(){
+    console.log(`Hellooooo, i am ${this._name} and i am ${this._age}.`)
+  }
+}
+const person1 = new Person ("Merve",34, "female");
+person1.greet();
+
+class Student extends Person {
+    constructor(name,age,gender,major){
+        super(name,age,gender)
+        this._major = major;
+    }
+    greet(){
+        console.log(`Helloo, i am a ${this._major}`);
+    }
+}
+const student1 = new Student ("Merve", 34, "female","J. Developer");
+student1.greet();
+
+// 2. Create a Shape class with name and color properties, and a describe() method that logs a description
+// of the shape to the console (e.g. "This is a blue square").
+// Then create a Square class that extends the Shape class and adds sideLength and area properties.
+// Override the describe() method in the Square class to log a description of the square to the console (e.g. "This is a blue square with a side length of 5 and an area of 25").
+
+class Shape {
+    constructor(name,color){
+        this._name = name;
+        this._color = color;
+    }
+    describe(){
+        console.log(`This is ${this._color} ${this._name}.`);
+    }
+}
+const myShape = new Shape ("shape", "black");
+myShape.describe();
+
+class Square extends Shape{
+    constructor(name,color,sideLength){
+        super(name,color);
+        this._sideLength = sideLength;
+        this.area = sideLength * sideLength;
```

Comment:
Since you have other propertied inside classes defined with an underscore, make sure to use underscore in all places. Consistency is important :) 

```suggestion
        this._area = sideLength * sideLength;
```